neo/core/spiketrainlist.py

```diff
@@ -278,10 +298,14 @@ def _spiketrains_from_array(self):
                 mask = self._channel_id_array == channel_id
                 times = self._spike_time_array[mask]
                 spiketrain = SpikeTrain(times, **self._spiketrain_metadata)
-                spiketrain.annotations = {
-                    name: value[i]
-                    for name, value in self._annotations.items()
-                }
+                for name, value in self._annotations.items():
+                    if (not isinstance(value, str)
+                        and hasattr(value, "__len__")
+                        and len(value) == len(self._all_channel_ids)
+                       ):
+                        spiketrain.annotate(**{name: value[i]})
+                    else:
+                        spiketrain.annotate(**{name: value})
```

Comment:
This is introducing a kind of `array_annotation` across spiketrains. Does this imply you want to remove the array_annotation mechanism for single spiketrains in the long run?